File: src/main/java/org/example/libraryapp/errorHandler/BookAccessDeniedException.java
package org.example.libraryapp.errorHandler;

import org.springframework.http.HttpStatus;

public class BookAccessDeniedException extends BaseException {

    Long id;

    public BookAccessDeniedException(Long id) {
        super(HttpStatus.BAD_REQUEST, new ApiError(
                400,
                "Bad Request",
                "Book with id" + id + " not accessible for updates"));
        this.id = id;
    }
}

File: src/main/kotlin/org/example/libraryapp/errorHandler/BookAccessDeniedException.kt
package org.example.libraryapp.errorHandler

import org.springframework.http.HttpStatus

class BookAccessDeniedException(id: Long) : BaseException(
    HttpStatus.BAD_REQUEST,
    ApiError(
        400,
        "Bad Request",
        "Book with id $id not accessible for updates"
    )
)